src/rules/enforce-assert-throws.ts

```
@@ -25,6 +25,43 @@ export const enforceAssertThrows = createRule<[], MessageIds>({
+        // Check if it's an argument to a function call or the callee (IIFE)
+        if (parent.type === AST_NODE_TYPES.CallExpression) {
+          // Check if the CallExpression itself is part of a ternary or logical expression
+          const grandParent = parent.parent;
+          if (
+            grandParent &&
+            (grandParent.type === AST_NODE_TYPES.ConditionalExpression ||
+             grandParent.type === AST_NODE_TYPES.LogicalExpression)
+          ) {
+            return true;
+          }
+          return parent.arguments.includes(node as any) || parent.callee === node;
```

ESLint

[error] Replace `parent.arguments.includes(node·as·any)·||·parent.callee·===·node` with `(⏎············parent.arguments.includes(node·as·any)·||·parent.callee·===·node⏎··········)`

(prettier/prettier)